src/__test__/out_object.spec.js

```diff
@@ -28,46 +28,187 @@ describe('Command', () => {
+
+            it('works with input params', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
+
+                await page.property('onResourceRequested', true, function(requestData, networkRequest, test, out) {
+                    out.data = test;
+                }, 'test', outObj);
+
+                await page.open(`http://localhost:${port}/test2`);
+                let data = await outObj.property('data');
+                expect(data).toEqual('test');
+            });
+        });
+        describe('("onResourceReceived", true)', () => {
+            it('returns a value set by phantom', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
+
+                await page.property('onResourceReceived', true, function(response, out) {
+                    out.lastResponse = response;
+                }, outObj);
 
-        outObj.test = 'fooBar$';
+                await page.open(`http://localhost:${port}/test`);
 
-        await page.property('onResourceReceived', function(response, out) {
-            out.data = out.test + response.url;
-        }, outObj);
+                let lastResponse = await outObj.property('lastResponse');
 
-        await page.open(`http://localhost:${port}/test2`);
-        let data = await outObj.property('data');
-        expect(data).toEqual(`fooBar$http://localhost:${port}/test2`);
+                expect(lastResponse.url).toEqual(`http://localhost:${port}/test`);
+            });
+
+            it('returns a value set by phantom and node', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
+
+                outObj.test = 'fooBar$';
+
+                await page.property('onResourceReceived', true, function(response, out) {
+                    out.data = out.test + response.url;
+                }, outObj);
+
+                await page.open(`http://localhost:${port}/test2`);
+                let data = await outObj.property('data');
+                expect(data).toEqual(`fooBar$http://localhost:${port}/test2`);
+            });
+
+            it('works with input params', async () => {
+                let page = await phantom.createPage();
+                let outObj = phantom.createOutObject();
+
+                await page.property('onResourceReceived', true, function(response, test, out) {
+                    out.data = test;
+                }, 'test', outObj);
+
+                await page.open(`http://localhost:${port}/test2`);
+                let data = await outObj.property('data');
+                expect(data).toEqual('test');
+            });
+        });
     });
 
-    it('#property() works with input params', async () => {
-        let page = await phantom.createPage();
-        let outObj = phantom.createOutObject();
+    describe('#property', () => {
```

Comment:
the Property specs here in this describe block pass.  All other specs in this change set fail.